File: leetcode/easy/java/lc94.java
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> results = new ArrayList<>();
        traverse(root, results);
        return results;
    }

    private void traverse(TreeNode root, List<Integer> results) {
        if (root == null) {
            return;
        }
        traverse(root.left, results);
        results.add(root.val);
        traverse(root.right, results);
    }
}

File: leetcode/easy/go/lc94.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val   int
 *     Left  *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
    traversedTree := []int{}
    traverse(root, &traversedTree)
    return traversedTree
}

func traverse(root *TreeNode, traversedTree *[]int) {
    if root == nil {
        return
    }
    traverse(root.Left, traversedTree)
    *traversedTree = append(*traversedTree, root.Val)
    traverse(root.Right, traversedTree)
}

